verification/curator-service/api/src/geocoding/mapbox.ts

```diff
@@ -91,12 +91,20 @@ export default class MapboxGeocoder {
             // with features of that specific type.
             // The context of the feature can still has other types
             // (if you ask for a region, the context will still contain the country).
-            if (opts?.limitToResolution) {
-                const type = resolutionToGeocodeQueryType.get(
-                    opts?.limitToResolution,
-                );
-                if (type) {
-                    req.types = [type];
+            const requestedResolutions = opts?.limitToResolution;
+            if (requestedResolutions) {
+                const types = requestedResolutions
+                    .map((r: Resolution) => resolutionToGeocodeQueryType.get(r))
+                    .filter((t) => t);
+                if (types) {
+                    req.types = [];
```

Comment:
can you do the filter & assignment all in one again here `req.types = types.filter((t) => t)` or does it still give you type issues?

Reply:
I'd started as one line and expanded until Typescript took the hint. It seems that `(t) => t` (or even, strangely, `t !== undefined`) isn't sufficient to register the content of the array as `string`.